lib/system.nim

```diff
@@ -1522,10 +1522,27 @@ include "system/iterators_1"
 
 {.push stackTrace: off.}
 
-proc abs*(x: float64): float64 {.noSideEffect, inline.} =
-  if x < 0.0: -x else: x
-proc abs*(x: float32): float32 {.noSideEffect, inline.} =
-  if x < 0.0: -x else: x
+
+when defined(js):
+  proc fabs(x: float64): float64 {.importc: "Math.abs".}
+  proc fabsf(x: float32): float32 {.importc: "Math.abs".}
+else:
+  proc fabs(x: cdouble): cdouble {.importc: "fabs", header: "math.h".}
+  proc fabsf(x: cfloat): cfloat {.importc: "fabsf", header: "math.h".}
+
+proc abs*[T: float64 | float32](x: T): T {.noSideEffect, inline.} =
+  when nimvm:
+    # TODO register VM ops
+    if x < 0.0: result = -x
+    elif x > 0.0: result = x
+    elif x == 0.0: result = 0.0 # handle 0.0, -0.0
+    else: result = x # handle NaN
+  else:
+    when T is float64:
+      result = fabs(x)
+    else:
+      result = fabsf(x)
```

Comment:
```suggestion
when defined(js):
  proc js_abs[T: SomeNumber](x: T): T {.importc: "Math.abs".}
else:
  proc c_fabs(x: cdouble): cdouble {.importc: "fabs", header: "math.h".}
  proc c_fabsf(x: cfloat): cfloat {.importc: "fabsf", header: "math.h".}

proc abs*[T: float64 | float32](x: T): T {.noSideEffect, inline.} =
  when nimvm:
    if x < 0.0: result = -x
    elif x == 0.0: result = 0.0 # handle 0.0, -0.0
    else: result = x # handle NaN, > 0
  else:
    when defined(js): result = js_abs(x)
    else:
      when T is float64:
        result = c_fabs(x)
      else:
        result = c_fabsf(x)
```

made a few changes:
* 1 less branch
* no need for vmops here, I feel like in this case, it's fast enough in VM
* changed fabs (etc) to c_fabs
* made js code more explicit (avoiding conflation with c)